src/main/java/org/opentripplanner/middleware/triptracker/TripStatus.java

```diff
@@ -65,23 +64,24 @@ public enum TripStatus {
     /**
      * Define the trip status based on the traveler's current position compared to expected and nearest points on the trip.
      */
-    public static TripStatus getTripStatus(
-        Coordinates currentPosition,
-        Instant currentTime,
-        Leg expectedLeg,
-        Segment segmentFromTime,
-        Segment segmentFromPosition
-    ) {
-        if (expectedLeg != null) {
-            if (segmentFromTime != null && isWithinModeBoundary(currentPosition, segmentFromTime)) {
+    public static TripStatus getTripStatus(TravelerPosition travelerPosition) {
+        if (travelerPosition.expectedLeg != null) {
+            if (travelerPosition.legSegmentFromTime != null &&
+                isWithinModeBoundary(travelerPosition.currentPosition, travelerPosition.legSegmentFromTime)
+            ) {
                 return TripStatus.ON_SCHEDULE;
             }
-            if (segmentFromPosition != null && isWithinModeBoundary(currentPosition, segmentFromPosition)) {
-                Instant segmentCenterTime = expectedLeg
+            if (travelerPosition.legSegmentFromPosition != null &&
```

Comment:
@binh-dam-ibigroup This makes sense and I've made the change. It does mean that `legSegmentFromTime` will be redundant because a lat/lon will always be attributed to a leg segment regardless of how far it is from it. Any thoughts on this before I remove the related code?

Reply:
I think it will be fine to remove.